### PR TITLE
Add mix functions for interleaved stereo samples.

### LINKED ISSUE #588: [Bug] Psirius Memories of Time differs between OMPT and libxmp
This have playback issues between OMPT and Libxmp with weird artifact plays on Ord 1 on libxmp, where OpenMPT doesn't.

[Modarchive](https://modarchive.org/index.php?request=view_by_moduleid&query=185766)

### PR DESCRIPTION
**Pull request by AliceLR**
Patch ~~4~~ 5 for stereo sample support. Adds and enables interleaved stereo sample mixers, and updates the loop wraparound smoothing code to correctly handle stereo samples. I reorganized the start of mixer.c slightly as well.

Closes #588 and #600.

Marked as a draft pending the regression tests, which are going to blow up the line count of this PR quite a bit. Feel free to review my progress as-is.

### CODE REVIEW

**Comment by AliceLR** [comment]
Okay, I'm (sort of) certain I didn't screw anything up here. I left it as two separate commits if you don't want to sift through the regression test junk.

**Review by sezero — APPROVED** [review]
OK with me. Waiting for @sagamusix as you requested.

File: src/mixer.c [hunk 1]
```diff
@@ -59,33 +54,52 @@ struct loop_data
...
+ * bit 2: 0=mono output, 1=stereo output
+ * bit 3: 0=unfiltered, 1=filtered
+ */
+
+#define FLAG_16_BITS	0x01
+#define FLAG_STEREO	0x02
+#define FLAG_STEREOOUT	0x04
+#define FLAG_FILTER	0x08
+#define FLAG_ACTIVE	0x10
+/* #define FLAG_SYNTH	0x20 */
```
Comment (sagamusix) [inline src/mixer.c:70]: Maybe it's time to retire this flag, as there is absolutely no indication what it's even supposed to do.

Comment (AliceLR) [reply src/mixer.c:70]: I have no clue either—I *assume* it's meant to signal to the mixer that some notes shouldn't be played as samples, but instead should be passed to a synthesizer. It seems like a mistake to me to send notes/effects to the synthesizer *after* the `read_event` functions and player.c have processed them but whatever.

I was just going to leave figuring this flag out until I add a new OPL player, whenever that is.

Comment (sagamusix) [reply src/mixer.c:70]: Looking back at a really old code revision, this does indeed seem to be the only remnant from OPL emulation. Maybe it has to be this way in libxmp but in OpenMPT such voices wouldn't even make their way into the mixer, so this seems strange to me.

File: src/mixer.c [hunk 2]
```diff
@@ -95,79 +109,45 @@ MIX_FN(stereoout_mono_a500);
...
-	libxmp_mix_monoout_mono_8bit_spline_filter,
-	libxmp_mix_monoout_mono_16bit_spline_filter,
-	libxmp_mix_stereoout_mono_8bit_spline_filter,
-	libxmp_mix_stereoout_mono_16bit_spline_filter
+	LIST_MIX_FUNCTIONS(spline_filter)
 #endif
 };
 
 #ifdef LIBXMP_PAULA_SIMULATOR
+#define LIST_MIX_FUNCTIONS_PAULA(type) \
```
[the remark this answers is not in the document]
Comment (AliceLR) [reply src/mixer.c:139]: > Does libxmp allow the Paula filter to be used with MED modules?

Doesn't look like it: all Paula mixer handling is gated with `IS_AMIGA_MOD()`, which requires both `m->read_event_type == READ_EVENT_MOD` and `m->period_type == PERIOD_MODRNG`. Supporting it for MED sounds reasonable but also like a separate issue; aside from new mixers, something would also have to be done about the `PERIOD_MODRNG` restriction.

[the remark this answers is not in the document]
Comment (sagamusix) [reply src/mixer.c:139]: Alright, so no issue here (unless there is any other MOD-like format that could have 16-bit samples...)

**Review by sagamusix — APPROVED** [review]

**Comment by sezero** [comment]
Is there anything left to do with this patch?

**Comment by AliceLR** [comment]
> Is there anything left to do with this patch?

No, I don't think so.

**Comment by sezero** [comment]
OK, this is in.

**Pull request merged**